src/config/compiler.rs

```diff
@@ -23,20 +23,7 @@ pub fn compile(mut builder: ConfigBuilder) -> Result<(Config, Vec<String>), Vec<
     }
 
     if errors.is_empty() {
-        Ok((
-            Config {
-                global: builder.global,
-                #[cfg(feature = "api")]
-                api: builder.api,
-                healthchecks: builder.healthchecks,
-                sources: builder.sources,
-                sinks: builder.sinks,
-                transforms: builder.transforms,
-                tests: builder.tests,
-                expansions,
-            },
-            warnings,
-        ))
+        Ok((builder.into_config(expansions), warnings))
```

Comment:
The idea of the `compile` function is that it's the only way to get a `Config` from a `ConfigBuilder`, which helps ensure that all `Config` values are valid (i.e. have gone through all the checks in `compile`). Extracting a new function here breaks that, since users can now just call `into_config` anywhere and there will be no checks done.